src/store/metadata/stub/validation.py

```diff
@@ -0,0 +1,423 @@
+            version, "versions", schemas.Versions
+        )
+        _validate_version_in_edition(version, versions_in_editions)
+
+
+def validate_topics(datasets, topics, topics_in_editions):
+    _validate_resource_count(topics, "topics")
+    _validate_resource_identifier_type_and_schema(topics, "topics", schemas.Topics)
+    topic_ids = [topic["@id"] for topic in topics["topics"]]
+    for dataset in datasets["datasets"]:
+        for topic in dataset["topics"]:
+            if topic not in topic_ids:
+                raise ValueError(f"{topic} not in list of approved topics {topic_ids}")
+    for topic in topics_in_editions:
+        if topic not in topic_ids:
+            raise ValueError(
+                f"""
+            Topic {topic} defined within edition but it is not (and must be) defined as a topic resource in topics.json.
+
+            Defined topics from topics.json:
+            {topic_ids}"""
+            )
+
+
+def validate_publishers(
+    publishers: Dict, dataset_publishers: Set, dataset_creators: Set
+):
+    """
+    Confirm that:
+    - any publisher listed at the `datasets` level exists
+    - any creator listed at the `datasets` level exists
+    - all publisher resources appear in one or more of the above checks
+    """
+    _validate_resource_count(publishers, "publishers")
+    publisher_ids = [publisher["@id"] for publisher in publishers["publishers"]]
+    publishers_referenced = []
+
+    for publisher in dataset_publishers:
+        if publisher not in publisher_ids:
+            raise ValueError(
+                f"""
+            The publisher: {publisher}
+            is referenced in a datasets resource but does not exist in the definitions of publishers from publishers.json.
+            """
+            )
+        publishers_referenced.append(publisher)
+
+    for creator in dataset_creators:
+        if creator not in publisher_ids:
+            raise ValueError(
+                f"""
+            The creator {creator} is referenced in a datasets resource but does not exist in the definitions of publishers from publishers.json.
+            """
+            )
+        publishers_referenced.append(creator)
+
+    # At this point all publishers we've defined should have appeared at least once.
+    unique_publishers_referenced = list(set(publishers_referenced))
+
+    unique_publishers_referenced.sort()
+    publisher_ids.sort()
+
+    if len(unique_publishers_referenced) != len(publisher_ids):
+        raise ValueError(
+            f"""
+        Publishers are being created that do not appear to be utilised.
+
+        IDs of publisher resources being created:
+        {publisher_ids}
+
+        IDs of publisher resources being used:
+        {unique_publishers_referenced}
+        """
+        )
+
+    # Schema validation
+    schemas.Publishers(**publishers)
+
+
+def _assert_editions_ordered_by_issued(list_of_datasets, main_dict: str, sub_dict: str):
```

Comment:
I dont think this is getting called anywhere. If I'm following this should be getting called in `validate_datasets()`

Reply:
Will delete - I rewrote it in _validate_edition_in_dataset L100-L109